src/io/record-play-movie.cpp

```diff
@@ -169,54 +170,50 @@ static bool string_is_repeat(char *str, int len)
 
 static errr send_text_to_chuukei_server(TERM_LEN x, TERM_LEN y, int len, TERM_COLOR col, concptr str)
 {
-    char buf[1024 + 32];
-    char buf2[1024];
-
-    strncpy(buf2, str, len);
-    buf2[len] = '\0';
-
     if (len == 1) {
-        sprintf(buf, "s%c%c%c%c", x + 1, y + 1, col, buf2[0]);
-    } else if (string_is_repeat(buf2, len)) {
-        int i;
-        for (i = len; i > 0; i -= 127) {
-            sprintf(buf, "n%c%c%c%c%c", x + 1, y + 1, std::min<int>(i, 127), col, buf2[0]);
+        insert_ringbuf(std::string("s").append(1, x + 1).append(1, y + 1).append(1, col).append(1, *str));
```

Comment:
Since the argument of insert_ringbuf funciont is `const std::string&`, it is safe to pass the return value of format directly.
For the sake of readability, it is better to pass the return value of format, which preserves the original sprintf format.
Also, if you use append, Visual Studio will give you a warning about narrowcasting from int to char at the `append(1, len)` point.

Reply:
Unless something higher in the call stack is holding the pointer returned by format() and is expecting the buffer to not be overwritten.  That's not likely with the text, wipe, and cursor hooks but might happen with the xtra hook since something calling term_xtra() might not expect that it would involve calling format() as a side effect.  I'll try to redo this pull request with something more readable but which still avoids format().